### ISSUE TITLE
[ecmascript] InnaccurateNumericLiteral false positive with scientific notations

### ISSUE THREAD

**Issue by Araxeus** [pmd/pmd#4345]
**Affects PMD Version:**  6.53.0

**Rule:** Error Prone: [InnaccurateNumericLiteral](https://pmd.sourceforge.io/pmd-6.5.0/pmd_rules_ecmascript_errorprone.html#innaccuratenumericliteral)

**Description:**

**Code Sample demonstrating the issue:**

```js
	const x = 1e3;
```

**Running PMD through:** *Codacy*

![image](https://user-images.githubusercontent.com/78568641/213679394-b04b5035-01ed-40b5-a42b-aaccf332cfbb.png)

**Comment by fudianchn** [comment]
Verified on 6.53.0 (the reported version), 7.3.0, 7.4.0 and current main (`6dbb11d`):

| input | 6.53.0 | 7.3.0 | 7.4.0 | main |
|---|---|---|---|---|
| `const x = 1e3;` (this report) | reported | reported | no violation | no violation |
| `999999999999999999999`, `1.12345678901234567` | reported | reported | reported | reported |
| `1.12e-4`, `999999999999999` | not reported | not reported | not reported | not reported |

The false positive is gone since 7.4.0, together with the rule rework that renamed it to InaccurateNumericLiteral (the old name is kept as a deprecated alias). b176ca72c1 determines inaccuracy by comparing the literal against its round-tripped double value, so exactly representable literals like `1e3` are no longer reported while imprecise ones still are. Can this be closed?

**Comment by adangel** [comment]
See also #4716
